### DISCUSSION TITLE
Where does mise get the default path from?

### DISCUSSION THREAD

**Discussion by pjaspers** [jdx/mise#2535] [Q&A] [answered]
(On a mac) After `mise activate zsh` has been run, my `$PATH` has these in it:

```
/Users/pjaspers/.local/share/mise/installs/ruby/3/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:
```

Where is it finding those from? (mainly the apple ones) My `__MISE_ORIG_PATH` doesn't have any of these, and I would like to avoid putting in `/usr/bin/` in front of my original path (because then i have old system git instead of homebrew git)

**Comment by Marukome0743** [comment, marked as answer]
Those Apple-specific PATH entries are normally added by macOS rather than by mise.

On macOS, `/etc/zprofile` runs `/usr/libexec/path_helper`, which constructs PATH from `/etc/paths` and files under `/etc/paths.d`. The entries shown here, including `/System/Cryptexes/...`, `/var/run/com.apple.security.cryptexd/...`, and `/Library/Apple/usr/bin`, match the output produced by `path_helper`.

I verified this in an isolated login zsh session. Starting with only:

```text
/opt/homebrew/bin:/usr/bin:/bin
```

the login shell added the same Apple paths before mise was activated. On the current main branch, `mise activate zsh` then preserved that PATH in `__MISE_ORIG_PATH`; mise did not generate the Apple entries itself.

There have also been subsequent fixes for PATH changes that happen after activation:

- https://github.com/jdx/mise/pull/6755 makes the first zsh prompt re-evaluate PATH changes made during shell initialization, including `path_helper`-style changes.
- https://github.com/jdx/mise/pull/8190 preserves PATH reordering performed after activation, such as changes made from `.zlogin`.

The exact reason the reported PATH differed from `__MISE_ORIG_PATH` likely depends on the order of the user’s zsh startup files: mise may have captured PATH first, and a later startup file or plugin may then have run `path_helper` or otherwise modified PATH.

To keep Homebrew’s Git ahead of `/usr/bin/git`, ensure Homebrew’s shell environment is applied in the appropriate startup file and before mise activation, for example:

```zsh
eval "$(/opt/homebrew/bin/brew shellenv)"
eval "$(mise activate zsh)"
```

With a current mise release, later intentional PATH reordering should also be preserved by `hook-env`.

*AI-assisted — Tool: Codex; model: OpenAI/GPT-5; version: unavailable.*
